Test is broken beginning with python 3.8.1

It seems in patch version of python 3.8.1 something changed underline.
Perhaps it was `urlparse`? Essentially schema is now being parsed different way, so `urlparse` no longer allows to put url without `http://` while also having port.
Obviously if test failed, its possible that actual codebase is failing now as well for certain scenarios (like one mentioned in #67).

This is the error i get on python 3.8.1 but not on any other version of python. Even python 3.8.0 doesn't have it (thats why we don't see right now TravisCI runner crashing this test).
```
yarn_api_client/base.py:82: in request
    response = self.session.request(method=method, url=api_endpoint, headers=headers, timeout=self.timeout, **kwargs)
/usr/local/lib/python3.8/site-packages/requests/sessions.py:533: in request
    resp = self.send(prep, **send_kwargs)
/usr/local/lib/python3.8/site-packages/requests_mock/mocker.py:111: in _fake_send
    return _original_send(session, request, **kwargs)
/usr/local/lib/python3.8/site-packages/requests/sessions.py:646: in send
    r = adapter.send(request, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <requests_mock.adapter.Adapter object at 0x7f6311890520>, request = <requests_mock.request._RequestObjectProxy object at 0x7f6311850070>
kwargs = {'cert': None, 'proxies': OrderedDict(), 'stream': False, 'timeout': 0, ...}, matcher = <requests_mock.adapter._Matcher object at 0x7f63119c7fa0>, resp = None

    def send(self, request, **kwargs):
        request = _RequestObjectProxy(request,
                                      case_sensitive=self._case_sensitive,
                                      **kwargs)
        self._add_to_history(request)

        for matcher in reversed(self._matchers):
            try:
                resp = matcher(request)
            except Exception:
                request._matcher = weakref.ref(matcher)
                raise

            if resp is not None:
                request._matcher = weakref.ref(matcher)
                resp.connection = self
                logger.debug('{} {} {}'.format(request._request.method,request._request.url, resp.status_code))
                return resp

>       raise exceptions.NoMockAddress(request)
E       requests_mock.exceptions.NoMockAddress: No mock address: GET example.com://80/ololo

/usr/local/lib/python3.8/site-packages/requests_mock/adapter.py:255: NoMockAddress
```

--- a/yarn_api_client/base.py
+++ b/yarn_api_client/base.py
@@ -25,6 +25,9 @@ class Response(object):
 
 class Uri(object):
     def __init__(self, service_endpoint):
+        if not (service_endpoint.startswith("http") and service_endpoint.startswith("https")):
+            service_endpoint = "http://" + service_endpoint
+
         service_uri = urlparse(service_endpoint)
         self.scheme = service_uri.scheme or 'http'
         self.hostname = service_uri.hostname or service_uri.path

